File: packages/jsii-pacmak/test/expected.java.jsii-calc/java/org/jsii/tests/calculator/UnaryOperation.java
package org.jsii.tests.calculator;
/**
 * An operation on a single operand.
 */
@org.jsii.Jsii(module = org.jsii.tests.calculator.$Module.class, fqn = "jsii-calc.UnaryOperation")
public abstract class UnaryOperation extends org.jsii.tests.calculator.lib.Operation {
    protected UnaryOperation(final org.jsii.JsiiObject.InitializationMode mode) {
        super(mode);
    }
    public org.jsii.tests.calculator.lib.Value getOperand() {
        return this.jsiiGet("operand", org.jsii.tests.calculator.lib.Value.class);
    }
}

File: packages/jsii-pacmak/test/expected.jsii-calc/dotnet/AWS.Jsii.Tests.Calculator/UnaryOperation.cs
using AWS.Jsii.Runtime.Deputy;
using AWS.Jsii.Tests.Calculator.Lib;

namespace AWS.Jsii.Tests.Calculator
{
    /// <summary>An operation on a single operand.</summary>
    [JsiiClass(typeof(UnaryOperation), "jsii-calc.UnaryOperation", "[{\"name\":\"operand\",\"type\":{\"fqn\":\"@scope/jsii-calc-lib.Value\"}}]")]
    public abstract class UnaryOperation : Operation
    {
        protected UnaryOperation(Value_ operand): base(new DeputyProps(new object[]{operand}))
        {
        }

        protected UnaryOperation(ByRefValue reference): base(reference)
        {
        }

        protected UnaryOperation(DeputyProps props): base(props)
        {
        }

        [JsiiProperty("operand", "{\"fqn\":\"@scope/jsii-calc-lib.Value\"}")]
        public virtual Value_ Operand
        {
            get => GetInstanceProperty<Value_>();
        }
    }
}